File: python/83 AbstractMethod.py
from abc import ABC, abstractmethod
import math
class Shape(ABC):
    def __init__(self,colour):
        super().__init__()
        self.colour=colour

    @abstractmethod
    def calculateArea(self):
        pass

    def __str__(self):
        return f"Colour is {self.colour}"
    
class Circle(Shape):
    def __init__(self,radius, colour="red"):
        super().__init__(colour)
        self.radius=radius
    
    def calculateArea(self):
        return math.pi*self.radius**2

class Square(Shape):
    def __init__(self,side, colour="black"):
        super().__init__(colour)
        self.side=side
    
    def calculateArea(self):
        return self.side**2
    
circle1 = Circle(5, "blue")
square1 = Square(10)
circle2 = Circle(2)

print(f"Area of {circle1.__class__.__name__}: {circle1.calculateArea():.2f}")
print(f"Area of {square1.__class__.__name__}: {square1.calculateArea():.2f}")
print(f"Area of {circle2.__class__.__name__}: {circle2.calculateArea():.2f}")

File: java/83 AbstractMethod.java
package code;
abstract class Shapes{
	String colour;

	public Shapes(String colour) {
		this.colour = colour;
	}
	abstract double calculateArea();
	
	String getColour() {
		return colour;
	}
	void setColour(String colour) {
		this.colour = colour;
	}
}
class Circle extends Shapes{
	double radius;

	public Circle(String colour,double radius) {
		super(colour);
		this.radius = radius;
	}

	@Override
	double calculateArea() {
		return Math.PI*Math.pow(this.radius,2);
	}
	
}
class Rectangle extends Shapes{
	double length,width;

	public Rectangle(String colour,double length, double width) {
		super(colour);
		this.length = length;
		this.width = width;
	}

	@Override
	double calculateArea() {
		return this.length*this.width;
	}
	
}
public class Test {
	public static void main(String[] args) {
		Circle circle1 = new Circle("Red", 5.0);
        Rectangle rectangle1 = new Rectangle("Blue", 4.0, 6.0);

        System.out.println("Area: " + circle1.calculateArea());

        System.out.println("Area: "+rectangle1.calculateArea());
	}
}